### ISSUE TITLE
Adding custom information to generated pact json file

### ISSUE THREAD

**Issue by ghost** [pact-foundation/pact-jvm#400]
Hi!

Is it somehow possible to add a custom metadata tag in the generated pact json file?

Something like this:
```
    "metadata": {
        "pact-specification": {
            "version": "2.0.0"
        },
        "pact-jvm": {
            "version": "3.3.6"
        },
        "internal-project-version": {
            "version": "34.A"
        }
    }
```

Yours
Carlo

**Comment by shashidesai** [comment]
we can add more tags in the `DEFAULT_METADATA` - but then you would have to compile and have your own jar version - or we could ask @uglyog if its fine and makes sense to add custom tags and I can put in a PR - https://github.com/DiUS/pact-jvm/blob/master/pact-jvm-model/src/main/groovy/au/com/dius/pact/model/BasePact.groovy#L18

**Comment by uglyog** [comment]
It definitely makes sense. The point of the metadata map is to be able record any key-value pairs. The only important information in there is the specification version.

**Comment by uglyog** [comment]
Released 4.3.4 with this change
